crates/ide/src/references.rs

```diff
@@ -29,7 +29,7 @@ use crate::{display::TryToNav, FilePosition, FileRange, NavigationTarget, RangeI
 #[derive(Debug, Clone)]
 pub struct ReferenceSearchResult {
     declaration: Declaration,
-    references: Vec<Reference>,
+    references: UsageSearchResult,
```

Comment:
Getting back to this, it is interesting. This module wants to poke the USR quite a bit. The reason for that is I think is https://github.com/rust-analyzer/rust-analyzer/issues/4290. `ReferenceSearchResult` should be a stupid POD
UI type, which is just converted to JSON and shipped to the editor. Instead, it is used by rename to drive the logic of computation. 

I think we need to change the rename to work with `ide_db/search` directly. After that, we should be able to privatize the `UsageSearchResult::references` and slim down `ide/references.rs`